nomad/structs/structs.go

```diff
@@ -1558,23 +1572,41 @@ func (p *PeriodicConfig) Validate() error {
 		return nil
 	}
 
+	var mErr multierror.Error
 	if p.Spec == "" {
-		return fmt.Errorf("Must specify a spec")
+		multierror.Append(&mErr, fmt.Errorf("Must specify a spec"))
+	}
+
+	// Check if we got a valid time zone
+	if p.TimeZone != "" {
+		if _, err := time.LoadLocation(p.TimeZone); err != nil {
+			multierror.Append(&mErr, fmt.Errorf("Invalid time zone %q: %v", p.TimeZone, err))
+		}
 	}
 
 	switch p.SpecType {
 	case PeriodicSpecCron:
 		// Validate the cron spec
 		if _, err := cronexpr.Parse(p.Spec); err != nil {
-			return fmt.Errorf("Invalid cron spec %q: %v", p.Spec, err)
+			multierror.Append(&mErr, fmt.Errorf("Invalid cron spec %q: %v", p.Spec, err))
 		}
 	case PeriodicSpecTest:
 		// No-op
 	default:
-		return fmt.Errorf("Unknown periodic specification type %q", p.SpecType)
+		multierror.Append(&mErr, fmt.Errorf("Unknown periodic specification type %q", p.SpecType))
 	}
 
-	return nil
+	return mErr.ErrorOrNil()
+}
+
+func (p *PeriodicConfig) Canonicalize() {
+	// Load the location
+	l, err := time.LoadLocation(p.TimeZone)
+	if err != nil {
+		panic(fmt.Sprintf("Bad location %q: %v", p.TimeZone, err))
```

Comment:
@briansorahan The panic acts as an assertion. If it is hit there is a bug in implementation. The reason it is okay is Validate should check for a valid timezone: https://github.com/hashicorp/nomad/pull/2321/files/f4cc46948f1893c3b4afc090c76f41abecefeb24#diff-396d9428285e4245041a3d0b15546d5dR1581

So the system should never even canonicalize an invalid spec.